### PR TITLE
[DDW-335]: Fix double loading spinner on stake pools list page

### PR DESCRIPTION
**Pull request by yakovkaravelov**
This PR fixed double loading spinner on stake pools list page.
[Jira Ticket](https://jira.iohk.io/browse/DDW-335)

**This PR also enables Yoroi "Shelley" wallets restoration.**

### Yoroi "Shelley" wallets restoration
Screenshot 2020-08-12 at 09 07 58
Screenshot 2020-08-12 at 09 09 50

## Testing Checklist
- [Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1597099837029400)

### App
- [x] Go to `Staking` => `Stake Pools` and make sure double loading spinner is never displayed.

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

**Comment by nikolaglumac** [comment]
@darko-mijic @gnpf @mioriohk @ManusMcCole can you please check if this works now?

**Comment by mioriohk** [comment]
Looks good to me. Unable to reproduce on Windows and Linux. @ManusMcCole unable to reproduce on Mac

File: source/renderer/app/containers/staking/StakePoolsListPage.js [hunk 1]
```diff
@@ -53,14 +53,15 @@ export default class StakePoolsListPage extends Component<Props> {
...
-      isRanking,
       selectedDelegationWalletId,
       stake,
       fetchingStakePoolsFailed,
       recentStakePools,
       getStakePoolById,
     } = staking;
     const { all } = wallets;
+    const isLoading = !isSynced || fetchingStakePoolsFailed;
+    const isRanking = staking.isRanking && stakePoolsRequest.isExecuting;
```
Comment (daniloprates) [inline source/renderer/app/containers/staking/StakePoolsListPage.js:64]: What about adding `!isLoading` here?

File: source/renderer/app/containers/staking/StakePoolsListPage.js [hunk 2]
```diff
@@ -75,11 +76,11 @@ export default class StakePoolsListPage extends Component<Props> {
...
           stake={stake}
           onDelegate={this.handleDelegate}
-          isLoading={!isSynced || fetchingStakePoolsFailed}
-          isRanking={isRanking && stakePoolsRequest.isExecuting}
+          isLoading={isLoading}
+          isRanking={isRanking}
           getStakePoolById={getStakePoolById}
         />
-        {isRanking && stakePoolsRequest.isExecuting && (
+        {!isLoading && isRanking && stakePoolsRequest.isExecuting && (
```
Comment (daniloprates) [inline source/renderer/app/containers/staking/StakePoolsListPage.js:83]: You can add `!isLoading` to the `isRanking` const and remove `stakePoolsRequest.isExecuting` from here, as it's already checked above.

File: source/renderer/app/containers/staking/StakePoolsListPage.js (hunk 1 above)
Comment (nikolaglumac) [reply source/renderer/app/containers/staking/StakePoolsListPage.js]: @daniloprates @yakovkaravelov resolved in: https://github.com/input-output-hk/daedalus/pull/2144/commits/e682113e7e5406b6ff60a064e6719facbb677699

File: source/renderer/app/containers/staking/StakePoolsListPage.js (hunk 2 above)
Comment (nikolaglumac) [reply source/renderer/app/containers/staking/StakePoolsListPage.js]: @daniloprates @yakovkaravelov resolved in: https://github.com/input-output-hk/daedalus/pull/2144/commits/e682113e7e5406b6ff60a064e6719facbb677699

**Pull request merged**